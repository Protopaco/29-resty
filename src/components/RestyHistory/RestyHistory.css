.history{
    border: 3px solid black;
    width: 300px;
    height: 72vh;
    font-size: .8em;
    overflow: scroll;
}

.history::-webkit-scrollbar {
    display: none;
  }

  .body{
      background: lightgray;
      border: .5px solid darkgray;
      width: 90%;
      margin-left: 7px;
      padding: 3px
  }