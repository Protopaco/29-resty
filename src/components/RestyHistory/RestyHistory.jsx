import React, { useState, useEffect } from 'react';
import styles from './RestyHistory.css'

const RestyHistory = ({ tempHistory }) => {
    const [history, setHistory] = useState();

    useEffect(() => {
        let fullHistory = JSON.parse(localStorage.getItem('history'));
        if (!fullHistory) fullHistory = [];
        if (fullHistory && tempHistory.url) fullHistory.push(tempHistory)

        localStorage.setItem('history', JSON.stringify(fullHistory))

        setHistory(fullHistory)
    }, [tempHistory])

    return (
        <div>
            <h2>HISTORY</h2>
            <div className={styles.history}>
                {
                    history ?
                        history.map((search, searchIndex) => {
                            return <div key={searchIndex}>
                                <p>{search.infoObject.method}</p>
                                <p>{search.url}</p>
                                {search.infoObject.body ?
                                    <div className={styles.body}>
                                        <p>BODY:</p>
                                        <p>{search.infoObject.body}</p>
                                    </div>
                                    :
                                    <></>
                                }
                                <hr />
                            </div>
                        })
                        :
                        <></>
                }
            </div >
        </div>
    )
}

export default RestyHistory;