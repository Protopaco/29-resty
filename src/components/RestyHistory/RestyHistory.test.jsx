import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RestyHistory from './RestyHistory.jsx';

describe('RestyHistory component', () => {
    afterEach(() => cleanup());
    it('renders RestyHistory', () => {
        const { asFragment } = render(<RestyHistory tempHistory={{}} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
