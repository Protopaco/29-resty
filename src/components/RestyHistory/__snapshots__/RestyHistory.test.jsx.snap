// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestyHistory component renders RestyHistory 1`] = `
<DocumentFragment>
  <div>
    <h2>
      HISTORY
    </h2>
    <div
      class="history"
    />
  </div>
</DocumentFragment>
`;
